gufe/protocols/protocolunit.py

```diff
@@ -49,7 +49,7 @@ def _list_dependencies(inputs, cls):
     return deps
 
 
-class ProtocolUnitResultBase(GufeTokenizable):
```

Comment:
I got rid of the base class as it was tiresome to type `Union[ProtocolUnitResult, ProtocolUnitFailure]` everywhere.  Now Failure is just a special case of Result